ec/ecresource/organizationresource/update.go

```diff
@@ -0,0 +1,291 @@
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"sort"
+	"strings"
+)
+
+func (r *Resource) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	diagnostics := &response.Diagnostics
+
+	var plan Organization
+	var state Organization
+	diagnostics.Append(request.Plan.Get(ctx, &plan)...)
+	diagnostics.Append(request.State.Get(ctx, &state)...)
+	if diagnostics.HasError() {
+		return
+	}
+
+	organizationID := plan.ID.ValueString()
+
+	planMembers := make(map[string]types.Object)
+	diags := plan.Members.ElementsAs(ctx, &planMembers, false)
+	if diags.HasError() {
+		diagnostics.Append(diags...)
+		return
+	}
+	stateMembers := make(map[string]types.Object)
+	diags = state.Members.ElementsAs(ctx, &stateMembers, false)
+	if diags.HasError() {
+		diagnostics.Append(diags...)
+		return
+	}
+
+	// Create new members, update changed members
+	for email, planMember := range planMembers {
+		planMemberModel := toModel(ctx, planMember, diagnostics)
+		if diagnostics.HasError() {
+			continue
+		}
+
+		// create new invitation if member is in plan but not in state
+		stateMember, ok := stateMembers[email]
+		if !ok {
+			r.createInvitation(ctx, email, planMemberModel, organizationID, diagnostics)
+		} else {
+			// member is in plan and state, update if there is a diff
+			if !stateMember.Equal(planMember) {
+				stateMemberModel := toModel(ctx, stateMember, diagnostics)
+				if diagnostics.HasError() {
+					continue
+				}
+				r.updateMember(ctx, stateMemberModel, planMemberModel, organizationID, diagnostics)
+			}
+		}
+	}
+
+	// Delete removed members
+	for key, stateMember := range stateMembers {
+		_, ok := planMembers[key]
+		if !ok {
+			// member is in state, but not in plan
+			stateMemberModel := toModel(ctx, stateMember, diagnostics)
+			if diagnostics.HasError() {
+				continue
+			}
+			r.deleteMember(stateMemberModel, organizationID, diagnostics)
+		}
+	}
+
+	// Re-read the whole org from the API to get the current state
+	updatedOrganization := r.readFromApi(ctx, organizationID, diagnostics)
+	if diagnostics.HasError() {
+		return
+	}
+	diagnostics.Append(response.State.Set(ctx, *updatedOrganization)...)
+}
+
+func (r *Resource) updateMember(
+	ctx context.Context,
+	stateMember OrganizationMember,
+	planMember OrganizationMember,
+	organizationID string,
+	diagnostics *diag.Diagnostics,
+) {
+	if planMember.InvitationPending.ValueBool() {
+		// Invitations can't be updated, so while the invitation is pending the role assignments can't be changed
+		// The only way to update them is by creating a new invitation with the right role-assignments.
+		r.deleteInvitation(planMember, organizationID, diagnostics)
+		r.createInvitation(ctx, stateMember.Email.ValueString(), planMember, organizationID, diagnostics)
```

Comment:
Just for my understanding: I guess `stateMember.Email` should equal to `planMember.Email`, correct?

Reply:
Yeah, the email is a computed read-only field which will never change. The plan is always the same as the state due to `UseStateForUnknown`. And it is always the same as the key in the map of members.

It would also be possible to always pass through the config-key of the map and use that. Since that is actually the source-of-truth for which member this is.

Reply:
I think using the same email (the one set by the user in the config) is the way to go. This way we always use the same source for all update calls (Instead of sometimes using the key and sometimes using the email field in the value of the map)